openedx_learning/core/publishing/models.py

```diff
@@ -13,36 +24,258 @@
+    PublishableEntity via a OneToOneField with primary_key=True. The easiest way
+    to do this is to have your model inherit from PublishableEntityMixin.
+
+    Identifiers
+    -----------
+    The UUID is globally unique and should be treated as immutable.
+
+    The identifier field *is* mutable, but changing it will affect all
+    PublishedEntityVersions. They are locally unique within the LearningPackage.
+
+    If you are referencing this model from within the same process, use a
+    foreign key to the id. If you are referencing this PublishedEntity from an
+    external system/service, use the UUID. The identifier is the part that is
+    most likely to be human-readable, and may be exported/copied, but try not to
+    rely on it, since this value may change.
+
+    Note: When we actually implement the ability to change identifiers, we
+    should make a history table and a modified attribute on this model.
+
+    """
+
+    uuid = immutable_uuid_field()
+    learning_package = models.ForeignKey(LearningPackage, on_delete=models.CASCADE)
+    identifier = identifier_field()
+    created = manual_date_time_field()
+    created_by = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+    )
+
+    class Meta:
+        constraints = [
+            # Identifiers are unique within a given LearningPackage.
+            models.UniqueConstraint(
+                fields=[
+                    "learning_package",
+                    "identifier",
+                ],
+                name="pub_ent_uniq_lp_identifier",
+            )
+        ]
+        indexes = [
+            # Global Identifier Index:
+            #   * Search by identifier across all PublishableEntities on the
+            #     site. This would be a support-oriented tool from Django Admin.
+            models.Index(
+                fields=["identifier"],
+                name="pub_ent_idx_identifier",
+            ),
+            # LearningPackage (reverse) Created Index:
+            #  * Search for most recently *created* PublishableEntities for a
+            #    given LearningPackage, since they're the most likely to be
+            #    actively worked on.
+            models.Index(
+                fields=["learning_package", "-created"],
+                name="pub_ent_idx_lp_rcreated",
+            ),
+        ]
+        # These are for the Django Admin UI.
+        verbose_name = "Publishable Entity"
+        verbose_name_plural = "Publishable Entities"
+
+    def __str__(self):
+        return f"{self.identifier}"
+
+
+class PublishableEntityVersion(models.Model):
+    """
+    A particular version of a PublishableEntity.
+
+    This model has its own ``uuid`` so that it can be referenced directly. The
+    ``uuid`` should be treated as immutable.
+
+    PublishableEntityVersions are created once and never updated. So for
+    instance, the ``title`` should never be modified.
```

Comment:
So if I make 100 small edits (to a draft) without publishing, there will still be 100 `PublishableEntityVersion` rows created, but just one `Draft` row which gets updated each time, and no `PublishLog[Record]`/`Published` rows. Is that right?